smoke_test/remote/non_cratesio/cargo/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "compile_with_bazel"
+version = "0.1.0"
+authors = ["Bradlee Speice <bradlee.speice@gmail.com>"]
+
+[dependencies]
+rand = { git = "https://github.com/rust-lang-nursery/rand.git", tag = "0.4.1" }
+env_logger = { git = "https://github.com/sebasmagri/env_logger.git", tag = "v0.5.5" }
+# Note that we use a (slightly) outdated version of log; because env_logger resolves a version
```

Comment:
Can you go into more detail about this problem? I recognize that I probably missed this in the original PR in cargo-raze-examples, but this sets off alarm bells in my head. I wouldn't want to be resolving more than one version of log if I could help it.

---

After typing that, I came up with a theory. Is env_logger being too greedy about the minor version that it chooses? That sounds like a larger issue that we might want to be thinking about, though it's probably only an issue when you use `git` dependencies for common dependencies -- a practice I would __*very strongly discourage*__, so much so that I might prefer this example be changed to include a git dep that is a leaf dependency.

Reply:
Not exactly, no. The issue is that when `env_logger` requests `log` version `0.4.1`, it uses `log = "0.4.1"` telling `cargo` to pick it up as a crate.

When we later include (via git) a second `log` version 0.4.1, `cargo` (or `rustc`?) is none too pleased about having duplicate sources for the same dependency. This hack essentially lets `env_logger` pull in whatever versions of `log` it wants for itself, and we'll take an older version (since a newer-er version doesn't exist yet) to get Cargo to stop complaining.

Reply:
**_TL;DR: LGTM for now, I'm concerned about not solving the different dependency problem and want to look at it in the future_**

IIUC, this hack only works for the particular kind of relationship that `env_logger` and `log` have -- one where there is a weird indirection between users of the log macros and the actual logger.

If these dependencies were more conventional (say, we were talking about `nalgebra` and `ncollide`), we would be in a world of pain in the main crate the second we tried to use both crates in the same context, as ncollide would not be using the same nalgebra as the one we use. 

...I'm actually not convinced that what you have here works. I suspect log! messages in the root crate might not propagate to env_logger as they would need to jump the gap between whatever version of `log` main has, and whatever version of `log` `env_logger` has.

That said, this isn't a problem we need to solve right now -- I'm just worried that this will lead folks into a dependency trap.